crds/workspace.devfile.io_devworkspaces.yaml

```diff
@@ -552,6 +554,7 @@ spec:
                             mountSources:
                               type: boolean
                             sourceMapping:
+                              default: /projects
```

Comment:
Do we really want here to have that default value? Then it will provide explicit value into devfile when user uses competition
![Screenshot_20201006_103838](https://user-images.githubusercontent.com/5887312/95172666-53719500-07c0-11eb-814d-286ec092a244.png)

I wonder if it makes sense to omit by purpose default values for things we don't expect to configure very often but that's an assumption that may be totally wrong


P.S. I got default fields filled in only with local VSCode, Che devfile/api workspace seems to comple only container field name but not more, not sure what is the problem:
![Screenshot_20201006_104445](https://user-images.githubusercontent.com/5887312/95173243-15c13c00-07c1-11eb-8930-3dc6fa60e2a5.png)

Reply:
P.S. I got default fields filled in only with local VSCode, Che devfile/api workspace seems to comple only container field name but not more, not sure what is the problem:
![Screenshot_20201006_104445](https://user-images.githubusercontent.com/5887312/95173243-15c13c00-07c1-11eb-8930-3dc6fa60e2a5.png)

Editor validation against schema does not seem to work as well
![Screenshot_20201006_104732](https://user-images.githubusercontent.com/5887312/95173525-751f4c00-07c1-11eb-9dae-a10595ca259b.png)

Reply:
Thanks @sleshchenko I'll have a look to those problems.

Reply:
Oh, OK, I think I know the reason: it takes the default devfile 1.0 schema available in the schema store instead of the local one configured in the theia settings.json. I'll answer in the other issue.

Reply:
> Do we really want here to have that default value? Then it will provide explicit value into devfile when user uses competition

Oh, I didn't realize that the yaml completion was automatically adding the defaults when doing object completion. That seems weird, but isn't it more a problem of the yaml language server ? 

> 
> I wonder if it makes sense to omit by purpose default values for things we don't expect to configure very often but that's an assumption that may be totally wrong

Well, I don't really know where to draw the line here, between "very often" and "not very often".

@l0rd @gorkem What do you think about that ? Do we still want to include default values in the Json schema itself instead of simply documenting them ?

Reply:
> That seems weird, but isn't it more a problem of the yaml language server ?

I'm not sure it's weird.
The default value is a value but not default behavior, they are a bit different things. I think such behavior works fine in a combination of the required field and defaultValue.
We might be interesting to propose users like 256Mi for memory limit to easier start from scratch and show the example for format.
Exposure type is another thing that we may want for user to be clear about.

But I agree about 
> between "very often" and "not very often".
that may be difficult to formulate rule when a field should be promoted with a default value and when not

Reply:
> > That seems weird, but isn't it more a problem of the yaml language server ?
> 
> I'm not sure it's weird.
> The default value is a value but not default behavior, they are a bit different things. I think such behavior works fine in a combination of the required field and defaultValue.

I agree that if a field is mandatory and has a default value, it should be filled with the default value. But what if it is optional ? Why force the user to add it ?

Reply:
IMO after reviewing hundreds of different JSON schemas that the YAML Language Server uses the default value should be auto inserted. It is the default value that will be processed if a user doesn't change it. According to the JSON schema spec `JSON processing tools may use this information to provide a default value for a missing key/value pair`. Also, it is the convention used by the JSON Language Server as well. If you didn't want this item to autocomplete automatically it should use the `examples` tag instead

Reply:
@JPinkney I understand your point about the expected YAML language server behavior. However I'm not sure that, for the devfile, we have a strong agreement that we should force all the fields with a default value to be added automatically by completion *in all cases*.

So for now I just removed the `default` directives from the Json schemas that are IDE-targeted. But of course default values are still described in the docs.

Reply:
Yes, but only with the `v1` apiextensions API: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting